File: snippets/python/src/list_payments.py
from breez_sdk_liquid import BindingLiquidSdk, GetPaymentRequest, ListPaymentDetails, ListPaymentsRequest, PaymentType
import logging


def get_payment(sdk: BindingLiquidSdk):
    try:
        # ANCHOR: get-payment
        payment_hash = "<payment hash>"
        sdk.get_payment(GetPaymentRequest.PAYMENT_HASH(payment_hash))

        swap_id = "<swap id>"
        sdk.get_payment(GetPaymentRequest.SWAP_ID(swap_id=swap_id))
        # ANCHOR_END: get-payment
    except Exception as error:
        logging.error(error)
        raise

def list_payments(sdk: BindingLiquidSdk):
    try:
        # ANCHOR: list-payments
        sdk.list_payments(ListPaymentsRequest())
        # ANCHOR_END: list-payments
    except Exception as error:
        logging.error(error)
        raise

def list_payments_filtered(sdk: BindingLiquidSdk):
    try:
        # ANCHOR: list-payments-filtered
        req = ListPaymentsRequest(
            filters = [PaymentType.SEND],
            from_timestamp = 1696880000, 
            to_timestamp = 1696959200, 
            offset = 0,
            limit = 50)
        sdk.list_payments(req)
        # ANCHOR_END: list-payments-filtered
    except Exception as error:
        logging.error(error)
        raise

def list_payments_details_address(sdk: BindingLiquidSdk):
    try:
        # ANCHOR: list-payments-details-address
        address = "<Bitcoin address>"
        req = ListPaymentsRequest(
            details = ListPaymentDetails.BITCOIN(address=address))
        sdk.list_payments(req)
        # ANCHOR_END: list-payments-details-address
    except Exception as error:
        logging.error(error)
        raise

def list_payments_details_destination(sdk: BindingLiquidSdk):
    try:
        # ANCHOR: list-payments-details-destination
        destination = "<Liquid BIP21 or address>"
        req = ListPaymentsRequest(
            details = ListPaymentDetails.LIQUID(destination=destination,asset_id=None))
        sdk.list_payments(req)
        # ANCHOR_END: list-payments-details-destination
    except Exception as error:
        logging.error(error)
        raise

File: snippets/csharp/ListPayments.cs
using Breez.Sdk.Liquid;

public class ListPaymentsSnippets
{
    public void GetPayment(BindingLiquidSdk sdk)
    {
        // ANCHOR: get-payment
        try
        {
            var paymentHash = "<payment hash>";
            var paymentByHash = sdk.GetPayment(
                new GetPaymentRequest.PaymentHash(paymentHash)
            );

            var swapId = "<swap id>";
            var paymentBySwapId = sdk.GetPayment(
                new GetPaymentRequest.SwapId(swapId)
            );
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: get-payment
    }
    
    public void ListPayments(BindingLiquidSdk sdk)
    {
        // ANCHOR: list-payments
        try
        {
            var payments = sdk.ListPayments(new ListPaymentsRequest());
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: list-payments
    }

    public void ListPaymentsFiltered(BindingLiquidSdk sdk)
    {
        // ANCHOR: list-payments-filtered
        try
        {
            var payments = sdk.ListPayments(
                new ListPaymentsRequest(
                    new() { PaymentType.Send },
                    fromTimestamp: 1696880000,
                    toTimestamp: 1696959200,
                    offset: 0,
                    limit: 50
                ));
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: list-payments-filtered
    }

    public void ListPaymentsDetailsAddress(BindingLiquidSdk sdk)
    {
        // ANCHOR: list-payments-details-address
        try
        {
            var address = "<Bitcoin address>";
            var payments = sdk.ListPayments(
                new ListPaymentsRequest(
                    details: new ListPaymentDetails.Bitcoin(address)
                ));
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: list-payments-details-address
    }

    public void ListPaymentsDetailsDestination(BindingLiquidSdk sdk)
    {
        // ANCHOR: list-payments-details-destination
        try
        {
            var destination = "<Liquid BIP21 or address>";
            var payments = sdk.ListPayments(
                new ListPaymentsRequest(
                    details: new ListPaymentDetails.Liquid(assetId: null, destination: destination)
                ));
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: list-payments-details-destination
    }
}
